docs/source/cc_service.md

```diff
@@ -0,0 +1,297 @@
+
+The rest of this topic describes how to configure chaincode as an external service:
+
+* [Packaging chaincode](#packaging-chaincode)
+* [Configuring a peer to process external chaincode](#configuring-a-peer-to-process-external-chaincode)
+* [External builder and launcher sample scripts](#external-builder-and-launcher-sample-scripts)
+* [Writing chaincode to run as an external service](#writing-chaincode-to-run-as-an-external-service)
+* [Deploying the chaincode](#deploying-the-chaincode)
+* [Running the chaincode as an external service](#running-the-chaincode-as-an-external-service)
+
+## Packaging chaincode
+
+With the Fabric v2.0 chaincode lifecycle, chaincode is packaged and installed in a `.targz` format. The following `myccpackage.tgz` archive  demonstrates the required structure:
+
+```sh
+$ tar xvfz myccpackage.tgz
+code.tar.gz
+metadata.json
+```
+
+### code.tar.gz archive requirements
+
+The `code.tar.gz` archive must include a `connection.json` file which contains the chaincode endpoint and certificate information:
+
+* **address** - chaincode server endpoint accessible from peer. Must be specified in “<host>:<port>” format.
+* **Dial_timeout** - interval to wait for connection to complete. Specified as a string qualified with units (examples : "10s", "500ms", "1m"). Default is “3s” if not specified.
+* **tls_required** - true or false. If false "client_auth_required", "key_path", "cert_path", "root_cert_path" are not required. Default is “true”.
+* **client_auth_required** - if true then you need to specify "key_path" and "cert_path" for client authentication. Default is false. Ignored if tls_required is false.
+* **key_path** - path to the key file. This path is relative to the “release directory” (see “release” below). Required if client_auth_required is true. Ignored if tls_required is false.
+* **cert_path**  - path to the certificate file. This path is relative to the “release directory” (see “release” below). Must be specified if client_auth_required is true. Ignored if tls_required is false.
+* **root_cert_path**  - path to the root cert for authenticating the server. Required when tls_required is set to true.
+
+For example:
+```json
+{
+  "address": "your.chaincode.host.com:9999",
+  "dial_timeout": "10s",
+  "tls_required": true,
+  "client_auth_required": "true",
+  "key_path": "path/rooted/in/release/directory/key.pem",
+  "cert_path": "path/rooted/in/release/directory/key.pem",
+  "root_cert_path": "path/rooted/in/release/directory/key.pem"
+}
+```
+
+**Note:** The TLS files can be placed anywhere in the `code.tar.gz` archive because the directory of the contents of the `.targz` will be provided to the external chaincode builder scripts. The `/bin/release` script, described later in this document, should move the files to the appropriate paths relative to the release directory.
```

Comment:
Maybe add another note that there is a JIRA open https://jira.hyperledger.org/browse/FAB-17228 to remove these file components and have the PEM come in as part of the JSON. (So this will very likely change before release)

Reply:
In general we try not to link to Jiras in our doc or even mention future potential functionality because often it does not get implemented.

As a compromise I added - 
`In the future, the PEM may be included in the JSON itself. `

The thing is we need to remember to come back and update the doc around this.